lib/timers.js

```diff
@@ -1,52 +1,110 @@
 'use strict';
 
-const Timer = process.binding('timer_wrap').Timer;
+const TimerWrap = process.binding('timer_wrap').Timer;
 const L = require('internal/linkedlist');
-const assert = require('assert').ok;
+const assert = require('assert');
 const util = require('util');
 const debug = util.debuglog('timer');
-const kOnTimeout = Timer.kOnTimeout | 0;
+const kOnTimeout = TimerWrap.kOnTimeout | 0;
 
 // Timeout values > TIMEOUT_MAX are set to 1.
 const TIMEOUT_MAX = 2147483647; // 2^31-1
 
-// IDLE TIMEOUTS
+
+// HOW and WHY the timers implementation works the way it does.
+//
+// Since many parts of Node.js and user applications rely on timeouts, there may
```

Comment:
I would phrase it differently.

1. The reason why the timers implementation need to perform well is that it's used internally to handle _any_ http request/response or any TCP I/O, so it's always in the execution path for node programs that use at least one of these forms of I/O, and users can't opt out of that.

2. Because it needs to be performant, and because there's potentially a very large number of timers involved in a node program at any given time, their implementation needs to be able to handle a large number of them in a way that performs well.

If 1) wasn't true, it would still be nice to have timers perform well when there's a lot of them, but it wouldn't be as critical.